### PR TITLE
FEATURE: Allow non-uuid node identifiers

### PR DESCRIPTION
**Pull request by nezaniel**
This replaces all occurrences of node identifier validation against the UUID pattern with a validation against a less restrictive NodeIdentifier pattern

In addition to removing the restriction that characters have to appear in a specific order, it also allows all other lowercase characters.

### CODE REVIEW

**Comment by kdambekalns** [comment]
Original hints from updated description:

---

This can be tested using the current API and non-API calls during node creation and modification

- [x] Code follows the PSR-2 coding style
- [x] Tests have been created, run and adjusted as needed
- [x] The PR is created against dev-master, but be compatible with 3.0, too

File: Neos.Neos/Classes/Service/LinkingService.php [hunk 1]
```diff
@@ -64,7 +64,7 @@ class LinkingService
      *
      * @var string
      */
-    const PATTERN_SUPPORTED_URIS = '/(node|asset):\/\/(([a-f0-9]){8}-([a-f0-9]){4}-([a-f0-9]){4}-([a-f0-9]){4}-([a-f0-9]){12})/';
+    const PATTERN_SUPPORTED_URIS = '/(node|asset):\/\/([a-z0-9\-]+|([a-f0-9]){8}-([a-f0-9]){4}-([a-f0-9]){4}-([a-f0-9]){4}-([a-f0-9]){12})/';
```
Comment (kdambekalns) [inline Neos.Neos/Classes/Service/LinkingService.php]: The specific pattern for the UUID could be removed, it's implicit in the first pattern, isn't it?

**Comment by kdambekalns** [comment]
This has been lying around for quite a while, so maybe it could make it into a release this time? It looks pretty low risk, but might open the CR for new use-cases.

**Comment by kitsunet** [comment]
@gerhard-boden I would hit the merge button now, but wanted to give you a chance to veto...

**Comment by gerhard-boden** [comment]
@kitsunet Don't see any harm in this change, I'll merge now

**Comment by gerhard-boden** [comment]
Do we have a note anywhere in the docs that needs to be updated because of this?

**Pull request merged**